include/librealsense2/dds/dds-device-broadcaster.cpp

```diff
@@ -81,26 +81,27 @@ dds_device_broadcaster::dds_device_broadcaster( dds_participant & participant )
     , _new_client_handler( [this]( dispatcher::cancellable_timer timer ) {
         // We wait until the new reader callback indicate a new reader has joined or until the
         // active object is stopped
-        std::unique_lock< std::mutex > lock( _new_client_mutex );
-        _new_client_cv.wait( lock, [this, timer]() {
-            return _trigger_msg_send.load() || timer.was_stopped();
-        } );
-        if( _new_client_handler.is_active() && _trigger_msg_send.load() )
+        if( _active )
         {
-            _trigger_msg_send = false;
-            _dds_device_dispatcher.invoke( [this]( dispatcher::cancellable_timer ) {
-                for( auto sn_and_handle : _device_handle_by_sn )
-                {
-                    if( sn_and_handle.second.listener->_new_reader_joined )
+            std::unique_lock< std::mutex > lock( _new_client_mutex );
+            _new_client_cv.wait( lock, [this]() { return ! _active || _trigger_msg_send.load(); } );
+            if( _new_client_handler.is_active() && _trigger_msg_send.load() )
```

Comment:
Weird to use `_new_client_handler.is_active()` when the cv checked for `_active` -- confusing.

Reply:
Agreed, I will try to change and and use the better status function for this purpose.

Reply:
Done